account_invoice_actual_date/models/sale_order.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# Licence AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3.0).
+
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _prepare_invoice(self):
+        invoice_vals = super()._prepare_invoice()
+        if self.picking_ids:
```

Comment:
Do a proper job of identifying the right picking to use. `state` and `date_done` fields should be used in the method.
```suggestion
        picking = self._get_picking_for_actual_date()
        if picking:
```